### PR TITLE
test: tests for delete

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Move tests as well as prepare framework for delete tests

Describe your proposed changes here.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: query_tests/Cargo.toml [hunk 1]
```diff
@@ -10,19 +10,19 @@ description = "Tests of the query engine against different database configuratio
 [dependencies]
 async-trait = "0.1"
 data_types = { path = "../data_types" }
+datafusion = { path = "../datafusion" }
 once_cell = { version = "1.4.0", features = ["parking_lot"] }
+predicate = { path = "../predicate" }
```
Comment (NGA-TRAN) [inline query_tests/Cargo.toml:15]: These 2 are needed in scenarios.rs and its module is public. I do not know any better solution

File: server/src/db.rs [hunk 2]
```diff
@@ -3741,262 +3741,4 @@ mod tests {
             .await
             .unwrap();
     }
-
```
Comment (NGA-TRAN) [inline server/src/db.rs:3744]: The below deleted 2 tests are a part of the above added tests

File: query_tests/src/sql.rs [hunk 3]
```diff
@@ -808,3 +808,68 @@ async fn sql_select_all_different_tags_chunks() {
...
+        "| 2   | 1970-01-01T00:00:00.000000020Z |",
+        "+-----+--------------------------------+",
+    ];
+
+    // Data deleted when it is in MUB, and then moved to RUB and OS
+    run_sql_test_case!(
+        DeleteFromMubOneMeasurementOneChunk {},
+        "SELECT * from cpu",
+        &expected
+    );
```
Comment (NGA-TRAN) [inline query_tests/src/sql.rs:827]: This is equivalent with the removing `test_delete_on_different_chunk_types`

File: query_tests/src/sql.rs [hunk 4]
```diff
@@ -808,3 +808,68 @@ async fn sql_select_all_different_tags_chunks() {
...
+
+    // Data deleted when it is in RUB, and then moved OS
```
Comment (NGA-TRAN) [inline query_tests/src/sql.rs:829]: This is new

File: query_tests/src/sql.rs [hunk 5]
```diff
@@ -808,3 +808,68 @@ async fn sql_select_all_different_tags_chunks() {
...
+    run_sql_test_case!(
+        DeleteFromRubOneMeasurementOneChunk {},
+        "SELECT * from cpu",
+        &expected
+    );
+
+    // Data deleted when it is in OS
```
Comment (NGA-TRAN) [inline query_tests/src/sql.rs:836]: this is new

File: query_tests/src/sql.rs [hunk 6]
```diff
@@ -808,3 +808,68 @@ async fn sql_select_all_different_tags_chunks() {
...
+    run_sql_test_case!(
+        DeleteFromOsOneMeasurementOneChunk {},
+        "SELECT * from cpu",
+        &expected
+    );
+}
+
+#[tokio::test]
+async fn sql_select_with_deleted_data_from_multi_exprs() {
```
Comment (NGA-TRAN) [inline query_tests/src/sql.rs:845]: similarly to above function, the first case is equivalent to the removing `test_delete_on_different_chunk_types_multi_exprs` and the last 2 are new.

File: query_tests/Cargo.toml (hunk 1 above)
Comment (alamb) [reply query_tests/Cargo.toml:15]: I don't think these are actually new "real" dependencies for `query_test` - they are already used transitively by `query` so I don't think this appreciable change in overall dependencies

File: query_tests/src/scenarios.rs [hunk 7]
```diff
@@ -1,12 +1,14 @@
 //! This module contains testing scenarios for Db
-
+#[allow(unused_imports, dead_code, unused_macros)]
```
Comment (alamb) [inline query_tests/src/scenarios.rs:2]: why is this needed?

File: query_tests/src/scenarios.rs [hunk 8]
```diff
@@ -1198,3 +1200,648 @@ impl DbSetup for ChunkOrder {
...
+
+async fn make_delete_mub(lp_lines: Vec<&str>, pred: &Predicate) -> DbScenario {
+    let db = make_db().await.db;
+    // create an open MUB
+    write_lp(&db, &lp_lines.join("\n")).await;
+    // One open MUB, no RUB, no OS
+    assert_eq!(count_mutable_buffer_chunks(&db), 1);
+    assert_eq!(count_read_buffer_chunks(&db), 0);
+    assert_eq!(count_object_store_chunks(&db), 0);
+    // delete one row
```
Comment (alamb) [inline query_tests/src/scenarios.rs:1510]:
```suggestion
```

I don't think that comment is valid anymore (it may be deleting more than one row) -- likewise below there are several "delete one row" type comments which I think are stale

File: query_tests/src/scenarios.rs (hunk 7 above)
Comment (alamb) [reply query_tests/src/scenarios.rs:2]: I was able to remove them without issue

```diff
diff --git a/query_tests/src/scenarios.rs b/query_tests/src/scenarios.rs
index 5e6ac585..e8ce9cff 100644
--- a/query_tests/src/scenarios.rs
+++ b/query_tests/src/scenarios.rs
@@ -1,5 +1,4 @@
 //! This module contains testing scenarios for Db
-#[allow(unused_imports, dead_code, unused_macros)]
 use std::collections::HashMap;
 use std::sync::Arc;
 use std::time::{Duration, Instant};
diff --git a/server/src/db.rs b/server/src/db.rs
index 99f5dc13..e1ff050e 100644
--- a/server/src/db.rs
+++ b/server/src/db.rs
@@ -1301,9 +1301,7 @@ mod tests {
     use arrow::record_batch::RecordBatch;
     use bytes::Bytes;
     use chrono::{DateTime, TimeZone};
-    use datafusion::logical_plan::{col, lit};
     use futures::{stream, StreamExt, TryStreamExt};
-    use predicate::predicate::PredicateBuilder;
     use tokio_util::sync::CancellationToken;
 
     use ::test_helpers::assert_contains;
```

**Review by alamb — APPROVED** [review]
This is pretty epic @NGA-TRAN ! I think it would be good to remove the `#[allow(unused_imports, dead_code, unused_macros)]` prior to merging as I find those warnings have been quite helpful detecting bugs in other places

File: query_tests/src/scenarios.rs (hunk 7 above)
Comment (NGA-TRAN) [reply query_tests/src/scenarios.rs:2]: Nice. I will remove them

File: query_tests/src/scenarios.rs (hunk 8 above)
Comment (NGA-TRAN) [reply query_tests/src/scenarios.rs:1510]: Good catch. Copy and paste from old test :) I will remove them.
